rvc/lib/algorithm/synthesizers.py

```diff
@@ -76,28 +79,46 @@ def __init__(
         )
 
         if use_f0:
-            self.dec = GeneratorNSF(
-                inter_channels,
-                resblock_kernel_sizes,
-                resblock_dilation_sizes,
-                upsample_rates,
-                upsample_initial_channel,
-                upsample_kernel_sizes,
-                gin_channels=gin_channels,
-                sr=sr,
-                is_half=kwargs["is_half"],
-            )
+            if vocoder == "MRF HiFi-GAN":
+                self.dec = HiFiGAN(
+                    in_channel=inter_channels,
+                    upsample_initial_channel=upsample_initial_channel,
+                    upsample_rates=upsample_rates,
+                    upsample_kernel_sizes=upsample_kernel_sizes,
+                    resblock_kernel_sizes=resblock_kernel_sizes,
+                    resblock_dilations=resblock_dilation_sizes,
+                    gin_channels=gin_channels,
+                    sample_rate=sr,
+                    harmonic_num=8,
+                )
+            else:
+                self.dec = GeneratorNSF(
+                    inter_channels,
+                    resblock,
+                    resblock_kernel_sizes,
+                    resblock_dilation_sizes,
+                    upsample_rates,
+                    upsample_initial_channel,
+                    upsample_kernel_sizes,
+                    gin_channels=gin_channels,
+                    sr=sr,
+                    is_half=kwargs["is_half"],
+                )
         else:
-            self.dec = Generator(
-                inter_channels,
-                resblock_kernel_sizes,
-                resblock_dilation_sizes,
-                upsample_rates,
-                upsample_initial_channel,
-                upsample_kernel_sizes,
-                gin_channels=gin_channels,
-            )
-
+            if vocoder == "MRF HiFi-GAN":
+                print("MRF HiFi-GAN does not support training without pitch guidance.")
```

Comment:
How hard is to support it?

Reply:
It is not hard to enable or disable, but pitch guidance provides an important assistance to the generator in forming proper harmonics. We may be better off just disabling the outdated non-nsf training all together.

Besides it is also a question of making yet another pretrain set.